File: ruby/acronym/acronym.rb
class Acronym

  def self.abbreviate(msg)
    msg.gsub(/([^A-Z ])([A-Z])/,'\1 \2')
      .gsub(/-/,' ').split(' ')
      .compact.map{|w| w[0]}.join.upcase
  end
end

module BookKeeping
  VERSION= 1
end

File: javascript/acronym/acronym.js
var Acronyms = function() {}

Acronyms.parse = function(msg) {
  return msg
    .replace(/[a-z0-9]+([A-Z])/g, " $1")
    .split(/[- ]/)
    .map(function(word) {
      return word[0].toUpperCase();
    })
    .join('');

};

module.exports = Acronyms;
